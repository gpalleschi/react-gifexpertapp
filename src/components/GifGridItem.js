import React from 'react'
import PropTypes from 'prop-types'; // IMPORT


export const GifGridItem = ( {url, title} ) => {
  return (
    <div className="card animate__animated animate__fadein">
         <img src={url} alt={title}></img>
	 <p>{ title }</p>
    </div>
  )
}

// SOTTO IL PROP TYPES :
GifGridItem.propTypes = {

url : PropTypes.string.isRequired,
title : PropTypes.string.isRequired
}

export default GifGridItem;



