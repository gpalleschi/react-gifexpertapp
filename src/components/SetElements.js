import React from 'react';

import PropTypes from 'prop-types';

const SetElements = ({setElements}) => {

	const handleChange = (e) => {
		setElements(e.target.value)
	}

	return (<div className='elementSelection'>
	        <label for="elements">Gif Elements</label>
		<select ClassName='eComboBox' onChange={handleChange} name="elements" id="elements">
                        <option value="1">1</option>
                        <option value="5">5</option>
                        <option value="10" selected="selected">10</option>
                        <option value="15">15</option>
                        <option value="20">20</option>
                        <option value="30">30</option>
                        <option value="40">40</option>
                        <option value="50">50</option>
                        <option value="100">100</option>
                </select>
	        </div>
		);
}

SetElements.prototype = {
	setElements: PropTypes.func.isRequired
}

export default SetElements;