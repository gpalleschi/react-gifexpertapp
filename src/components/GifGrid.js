import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
//import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ( {category, elements} ) => {

  const { data:images, loading } = useFetchGifs( category, elements );

  return (
    <>
    <h3>{category}</h3>

    { loading && <p>Loading ...</p>}

    { 
    <div className="card-grid"> 
	 { images.map( img => <GifGridItem key={ img.id } { ...img }/> )}
    </div>
    }
    </>	  
  )
}


GifGrid.propTypes = {
  category : PropTypes.string.isRequired,
  elements : PropTypes.number.isRequired
}

export default GifGrid;