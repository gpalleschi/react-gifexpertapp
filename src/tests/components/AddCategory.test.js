import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';
import '@testing-library/jest-dom';

describe('Pruebas <AddCategory />', () => { 

	const setCategories = jest.fn();

	let wrapper = shallow( <AddCategory setCategories={setCategories}/> );

	beforeEach( () => {
		jest.clearAllMocks();
	        wrapper = shallow( <AddCategory setCategories={setCategories}/> );
	})

	test('Debe de mostrarse correctamente', () => {
		expect( wrapper ).toMatchSnapshot();
	})

	test('Debe de cambiar la caja de testo ', () => { 

		const input = wrapper.find('input');
		const value = 'Hola Mundo';
		input.simulate('change', { target : { value : value}});

	})

	test('No debe de postear la informacion on submit', () => {
		wrapper.find('form').simulate('submit',{ preventDefault: () => {}});

		expect( setCategories ).not.toHaveBeenCalled();
	 })


	 test('debe de llamar el setCategory y limpiar la caja de texto', () => {


		// 1) Simular el input change
		const input = wrapper.find('input');
		const value = 'Hola Mundo';
		input.simulate('change', { target : { value : value}});
		// 2) Simular el submit
		wrapper.find('form').simulate('submit',{ preventDefault (){}});
		// 3) setCategories se debe che haber llamado
		expect( setCategories ).toHaveBeenCalled();
		// Specifichiamo che viene chiamato una volta
		expect( setCategories ).toHaveBeenCalledTimes(1);
		// Chiamata con una funzione
		expect( setCategories ).toHaveBeenCalledWith( expect.any(Function));

		// 4) el valor de input deve essere vuoto
		expect(wrapper.find('input').prop('value')).toBe('');

	  })
})