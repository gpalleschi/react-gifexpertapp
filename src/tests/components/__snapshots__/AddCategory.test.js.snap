// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <AddCategory /> Debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="String to Search"
    type="Text"
    value=""
  />
</form>
`;
